alphatwirl_interface/config/binned_df_stage.py

```diff
@@ -0,0 +1,135 @@
+    binning = []
+    for i, one_bin_dimension in enumerate(bin_list):
+        if not isinstance(one_bin_dimension, dict):
+            raise BadBinnedDataframeConfig("binning item no. {} is not a dictionary".format(i))
+        cleaned_dimension_dict = {"_" + k: v for k, v in one_bin_dimension.items()}
+        binning.append(_create_one_dimension(stage_name, **cleaned_dimension_dict))
+    return binning
+
+
+def _create_one_dimension(stage_name, _in, _out, _bins=None, _index=None):
+    if not isinstance(_in, six.string_types):
+        msg = "{}: binning dictionary contains non-string value for 'in'"
+        raise BadBinnedDataframeConfig(msg.format(stage_name))
+    if not isinstance(_out, six.string_types):
+        msg = "{}: binning dictionary contains non-string value for 'out'"
+        raise BadBinnedDataframeConfig(msg.format(stage_name))
+    if _index and not isinstance(_index, six.string_types):
+        msg = "{}: binning dictionary contains non-string and non-integer value for 'index'"
+        raise BadBinnedDataframeConfig(msg.format(stage_name))
+
+    if _bins is None:
+        bin_obj = Echo(nextFunc=None)
+    elif isinstance(_bins, dict):
+        # - bins: {nbins: 6 , low: 1  , high: 5 , overflow: True}
+        # - bins: {edges: [0, 200., 900], overflow: True}
+        if "nbins" in _bins and "low" in _bins and "high" in _bins:
+            low = _bins["low"]
+            high = _bins["high"]
+            nbins = _bins["nbins"]
+            edges = np.linspace(low, high, nbins + 1)
+        elif "edges" in _bins:
+            edges = _bins["edges"]
+        else:
+            msg = "{}: No way to infer binning edges for in={}"
+            raise BadBinnedDataframeConfig(msg.format(stage_name, _in))
+        bin_obj = Binning(boundaries=edges)
+    else:
+        msg = "{}: bins is neither None nor a dictionary for in={}"
+        raise BadBinnedDataframeConfig(msg.format(stage_name, _in))
+
+    return (str(_in), str(_out), bin_obj, _index)
+
+
+def _create_weights(stage_name, weights):
+    if weights is None:
+        return None
+    if isinstance(weights, list):
+        return {str(w): _create_one_weight(stage_name, w) for w in weights}
+    elif isinstance(weights, dict):
+        return {k: _create_one_weight(stage_name, w) for k, w in weights.items()}
+    # else we've got a single, scalar value
+    return {"weighted": _create_one_weight(stage_name, weights)}
+
+
+def _create_one_weight(stage_name, weight):
+    if isinstance(weight, six.string_types):
+        # Assume this is an event attribute
+        return WeightCalculatorSingleAttr(weight)
+    elif isinstance(weight, (float, six.integer_types)):
+        return WeightCalculatorConst(weight)
+    raise BadBinnedDataframeConfig("{}: Unknown weight type {} (type={})".format(stage_name, weight, type(weight)))
+
+
+class WithInsertTableFileNameComposer():
```

Comment:
This class would be nice to move elsewhere, and ideally be incorporated (at least in functionality) into alphatwirl itself, since it's not really "interface" code, and is rather an extension to the standard `TableFileNameComposer` class.